### PR TITLE
Add private skill option while storing skills

### LINKED ISSUE #815: Add a parameter of "skill type" in skill data for each skill
**Actual Behaviour**

Currently, all the skills are public.

**Expected Behaviour**

In order to allow user to create skills for his/her web bot, user has to create private skills. These skills will be applicable only to the chatbot of the user.

There would be two types of skills-
- `Public` skills
- `Private` skills

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by DNS-404**
Fixes #815

Changes: Added a parameter `private`. It's a boolean which is `true` (yes) if the skill is private and `false` (no) if the skill is public.

### CODE REVIEW

File: src/ai/susi/mind/SusiSkill.java [hunk 1]
```diff
@@ -511,6 +518,7 @@ public static JSONObject getSkillMetadata(String model, String group, String lan
                     skillid.hasName(skillname)) {
 
                 skillMetadata.put("skill_name", skill.getSkillName() ==null ? JSONObject.NULL: skill.getSkillName());
+                skillMetadata.put("private_skill", skill.getPrivateSkill());
```
Comment (hkedia321) [inline src/ai/susi/mind/SusiSkill.java]: Can you try adding null check here as done for other parameters?

Comment (DNS-404) [reply src/ai/susi/mind/SusiSkill.java]: It's a `boolean` with initial value `false`. Null checks are for strings.

**Comment by DNS-404** [comment]
@hkedia321 @Akshat-Jain @PrP-11 @anshumanv @akshatnitd Please review.

File: src/ai/susi/mind/SusiSkill.java [hunk 2]
```diff
@@ -299,6 +301,10 @@ else if (bang_type.equals("console")) {
                     if(skillName.length() > 0)
                         json.put("skill_name",skillName);
                 }
+                if (line.startsWith("::private_skill") && (thenpos = line.indexOf(' ')) > 0) {
```
Comment (Orbiter) [inline src/ai/susi/mind/SusiSkill.java]: just call it "private", not "private_skill" because all modifiers change status of the skill somehow.

Comment (DNS-404) [reply src/ai/susi/mind/SusiSkill.java]: Okay

**Comment by DNS-404** [comment]
@Orbiter Done the changes. Please review

**Comment by DNS-404** [comment]
Created PR https://github.com/fossasia/susi_server/pull/855 after doing changes.

**Pull request closed without merging**